pkg/provisioners/gcppubsub/util/pubsub_wrapper.go

```diff
@@ -0,0 +1,221 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"context"
+	"errors"
+
+	"google.golang.org/api/option"
+
+	"golang.org/x/oauth2/google"
+
+	"cloud.google.com/go/pubsub"
+)
+
+// This file exists so that we can unit test failures with the PubSub client.
```

Comment:
This whole setup is a heroic effort to work around a library design that is impossible to test. Thanks for taking the time to make it work @Harwayne, but I wish you didn't have to. As good open source citizens, we should probably open an issue on the pubsub client repo describing the difficulty. :innocent: